gtk4/src/expression.rs

```diff
@@ -182,6 +182,108 @@ impl glib::value::ToValueOptional for Expression {
+    fn weak_property_expression(&self, property_name: &str) -> crate::PropertyExpression;
+}
+
+impl<T: IsA<glib::Object>> GObjectPropertyExpressionExt for T {
+    fn property_expression(&self, property_name: &str) -> crate::PropertyExpression {
+        let obj_expr = crate::ConstantExpression::new(self);
+        crate::PropertyExpression::new(T::static_type(), Some(&obj_expr), property_name)
+    }
+
+    fn weak_property_expression(&self, property_name: &str) -> crate::PropertyExpression {
+        let obj_expr = crate::ObjectExpression::new(self);
+        crate::PropertyExpression::new(T::static_type(), Some(&obj_expr), property_name)
+    }
+}
+
+// rustdoc-stripper-ignore-next
+/// Trait containing convenience methods to create chains of [`gtk::PropertyExpression`]
+/// or [`gtk::ClosureExpression`].
+///
+/// # Examples
+///
+/// Create an expression describing `list_item->item->number` and bind it to button's label.
+///
+/// ```
+/// let factory = SignalListItemFactory::new();
+/// factory.connect_setup(|_, list_item| {
+///     let button = Button::new(None);
+///     list_item.set_child(Some(&button));
+///
+///     list_item.property_expression("item")
+///         .property_expression("number")
+///         .bind(&button, "label", Some(&button));
+/// });
+/// ```
+///
+/// Same thing as above but adds a string prefix to the number.
+///
+/// ```
+/// let factory = SignalListItemFactory::new();
+/// factory.connect_setup(|_, list_item| {
+///     let button = Button::new(None);
+///     list_item.set_child(Some(&button));
+///
+///     list_item.property_expression("item")
+///         .property_expression("number")
+///         .closure_expression(|args| {
+///             let number: i32 = args[1].get().unwrap();
+///             format!("Number: {}", number)
+///         })
+///         .bind(&button, "label", Some(&button));
+/// });
+/// ```
+pub trait ExpressionChainExt {
+    // rustdoc-stripper-ignore-next
+    /// Create a [`gtk::PropertyExpression`] that looks up for `property_name`
+    /// with self as parameter
+    fn property_expression(&self, property_name: &str) -> crate::PropertyExpression;
+
+    // rustdoc-stripper-ignore-next
+    /// Create a [`gtk::ClosureExpression`] with self as a parameter
+    fn closure_expression<F, R>(self, f: F) -> crate::ClosureExpression
+    where
+        F: Fn(&[glib::Value]) -> R + 'static,
+        R: glib::value::ValueType;
+}
+
+impl<T: AsRef<Expression>> ExpressionChainExt for T {
+    fn property_expression(&self, property_name: &str) -> crate::PropertyExpression {
+        crate::PropertyExpression::new(self.as_ref().value_type(), Some(self), property_name)
+    }
+
+    fn closure_expression<F, R>(self, f: F) -> crate::ClosureExpression
+    where
+        F: Fn(&[glib::Value]) -> R + 'static,
+        R: glib::value::ValueType,
+    {
+        crate::ClosureExpression::new(&[self.as_ref().clone()], f)
```

Comment:
@bilelmoussaoui That function should probably take a `impl IntoIterator<Item = &'a Expression>` or so as argument instead of what it does now, but independent of this PR.

Reply:
oh indeed; @SeaDve do you think you can provide a PR for this? otherwise I will open an issue about it :)

Reply:
Sure